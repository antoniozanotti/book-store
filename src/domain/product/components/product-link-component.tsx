import { ProductLinkComponentInterface } from "./product-link-component-interface";
import { getAuthorsByProduct } from "../helpers/get-authors-by-product";
import Link from "next/link";

export function ProductLinkComponent({
  product,
  children,
  ...props
}: ProductLinkComponentInterface) {
  const link = `/product/${product.page.slug}`;
  const title = `${product.name} by ${getAuthorsByProduct(product)}`;
  
  return (
    <Link href={link} title={title} className={props.className}>
      {children}
    </Link>
  );
}
