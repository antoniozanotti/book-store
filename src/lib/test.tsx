import React, { ReactElement, ReactNode } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return <div data-testid="container">{children}</div>;
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
    notFound: () => null,
    usePathname: () => null,
    useSearchParams: () => null,
  };
});
