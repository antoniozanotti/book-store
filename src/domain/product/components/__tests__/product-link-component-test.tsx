import { render, screen } from "@/lib/test";
import { ProductLinkComponent } from "../product-link-component";
import { productWithPagEandBookMock } from "../../__tests__/product-with-page-and-book-mock";

describe("ProductLinkComponent", () => {
  it("should render", async () => {
    const { container } = render(<ProductLinkComponent product={productWithPagEandBookMock}>Test</ProductLinkComponent>);
    expect(container).toMatchSnapshot();
  });
});