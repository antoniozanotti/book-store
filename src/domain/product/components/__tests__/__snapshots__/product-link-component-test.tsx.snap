// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductLinkComponent should render 1`] = `
<div>
  <div
    data-testid="container"
  >
    <a
      href="/product/the-four-agreements-a-practical-guide-to-personal-freedom"
      title="The Four Agreements: A Practical Guide to Personal Freedom by Don Miguel Ruiz"
    >
      Test
    </a>
  </div>
</div>
`;
